File: ShapeButton/src/main/java/com/gcode/shapebutton/ShapeButtonConstant.java
package com.gcode.shapebutton;

import android.graphics.drawable.GradientDrawable;

import androidx.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * @OriginalAuthor: Vast Gui
 * @OriginalDate:
 * @EditAuthor: Vast Gui
 * @EditDate: 2021/11/12
 */
public class ShapeButtonConstant {

    public static final int OVAL_SHAPE = 1;
    public static final int RECT_SHAPE = 2;
    public static final int ROUNDED_RECT_SHAPE = 3;
    public static final int ANY_ROUNDED_RECT_SHAPE = 4;

    @Retention(RetentionPolicy.SOURCE)
    @IntDef({
            OVAL_SHAPE,RECT_SHAPE,ROUNDED_RECT_SHAPE,ANY_ROUNDED_RECT_SHAPE
    })
    public @interface ShapeType {}

    @Retention(RetentionPolicy.SOURCE)
    @IntDef({
            GradientDrawable.LINEAR_GRADIENT,
            GradientDrawable.RADIAL_GRADIENT,
            GradientDrawable.SWEEP_GRADIENT
    })
    public @interface GradientType {}
}

File: ShapeButton/src/main/kotlin/com/gcode/shapebutton/ShapeButtonConstant.kt
package com.gcode.shapebutton

import android.graphics.drawable.GradientDrawable
import androidx.annotation.IntDef

// Author: Vast Gui
// Email: guihy2019@gmail.com
// Date: 2021/11/12
// Description: 
// Documentation:

object ShapeButtonConstant {
    const val OVAL_SHAPE = 1
    const val RECT_SHAPE = 2
    const val ROUNDED_RECT_SHAPE = 3
    const val ANY_ROUNDED_RECT_SHAPE = 4

    @Retention(AnnotationRetention.SOURCE)
    @IntDef(OVAL_SHAPE, RECT_SHAPE, ROUNDED_RECT_SHAPE, ANY_ROUNDED_RECT_SHAPE)
    annotation class ShapeType

    @Retention(AnnotationRetention.SOURCE)
    @IntDef(
        GradientDrawable.LINEAR_GRADIENT,
        GradientDrawable.RADIAL_GRADIENT,
        GradientDrawable.SWEEP_GRADIENT
    )
    annotation class GradientType
}